lib/class-wp-theme-json-gutenberg.php

```diff
@@ -143,6 +144,10 @@ class WP_Theme_JSON_Gutenberg {
 	 *
 	 * - value_key     => the key that represents the value
 	 *
+	 * - value_func    => a function to generate the value
+	 *
+	 * - value_args    => array of keys that will be mapped to values passed to value_func
+	 *
```

Comment:
@oandregal I think this is a better option than `value_format` that we talked about before. It's more flexible and doesn't overload `value_key` as a string _or_ array.

`value_args` is slightly misleading since it's more like `key_args`, but calling it `value_args` makes it seem more like it's related to `value_func` and `value_key`.

Reply:
I'm not fully convinced about the approach. However, I'm open to land as it is and iterate, though.

Reply:
One simplification we can make is to remove the `value_args` and always pass two pieces of data to the callback: the `slug` and the values that we'll take from the `value_key`. We could also consider passing the `name` for completeness, although I'm not sure it's that useful.